core/storage/leveldb/leveldb.hpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_LEVELDB_HPP
+#define KAGOME_LEVELDB_HPP
+
+#include <leveldb/db.h>
+#include <leveldb/write_batch.h>
+#include "common/logger.hpp"
+#include "storage/buffer_map.hpp"
+
+namespace kagome::storage {
+
+  class LevelDB : public PersistedBufferMap {
+   public:
+    class Batch;
+    class Iterator;
+
+    ~LevelDB() override = default;
+
+    static outcome::result<std::unique_ptr<LevelDB>> create(
+        std::string_view path, leveldb::Options options = leveldb::Options(),
+        common::Logger logger = common::createLogger("leveldb"));
+
+    /**
+     * @brief Set read options, which are used in @see LevelDB#get
+     * @param ro options
+     */
+    void setReadOptions(leveldb::ReadOptions ro);
+
+    /**
+     * @brief Set write options, which are used in @see LevelDB#put
+     * @param wo options
+     */
+    void setWriteOptions(leveldb::WriteOptions wo);
+
+    std::unique_ptr<BufferMapIterator> iterator() override;
```

Comment:
comment other methods as well

Reply:
above two methods do not override anything, so I put comments there. Iterator method has docs in the parent interface